url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/domain/singleagent/sokoban2/Sokoban2Domain.java

```diff
@@ -0,0 +1,746 @@
+		s.addObject(o);
+		
+		Attribute dirAtt = o.getObjectClass().getAttribute(ATTDIR);
+		if(dirAtt != null){
+			o.setValue(ATTDIR, "south");
+		}
+		
+		return s;
+		
+	}
+	
+	public static State getClassicState(Domain domain){
+		
+		State s = getCleanState(domain, 3, 2, 1);
+		
+		setRoom(s, 0, 4, 0, 0, 8, "red");
+		setRoom(s, 1, 8, 0, 4, 4, "green");
+		setRoom(s, 2, 8, 4, 4, 8, "blue");
+		
+		setDoor(s, 0, 4, 6, 4, 6);
+		setDoor(s, 1, 4, 2, 4, 2);
+		
+		setAgent(s, 6, 6);
+		setBlock(s, 0, 2, 2, "basket", "red");
+		
+		
+		return s;
+		
+	}
+	
+	
+	public static void setAgent(State s, int x, int y){
+		ObjectInstance o = s.getFirstObjectOfClass(CLASSAGENT);
+		o.setValue(ATTX, x);
+		o.setValue(ATTY, y);
+	}
+	
+	public static void setAgent(State s, int x, int y, int dir){
+		ObjectInstance o = s.getFirstObjectOfClass(CLASSAGENT);
+		o.setValue(ATTX, x);
+		o.setValue(ATTY, y);
+		o.setValue(ATTDIR, dir);
+	}
+	
+	public static void setBlockPos(State s, int i, int x, int y){
+		ObjectInstance o = s.getObjectsOfTrueClass(CLASSBLOCK).get(i);
+		o.setValue(ATTX, x);
+		o.setValue(ATTY, y);
+	}
+	
+	public static void setBlock(State s, int i, int x, int y, String shape, String color){
+		ObjectInstance o = s.getObjectsOfTrueClass(CLASSBLOCK).get(i);
+		setBlock(o, x, y, shape, color);
+	}
+	
+	public static void setBlock(ObjectInstance o, int x, int y, String shape, String color){
+		o.setValue(ATTX, x);
+		o.setValue(ATTY, y);
+		o.setValue(ATTSHAPE, shape);
+		o.setValue(ATTCOLOR, color);
+	}
+	
+	public static void setRoom(State s, int i, int top, int left, int bottom, int right, String color){
+		ObjectInstance o = s.getObjectsOfTrueClass(CLASSROOM).get(i);
+		setRegion(o, top, left, bottom, right);
+		o.setValue(ATTCOLOR, color);
+	}
+	
+	public static void setDoor(State s, int i, int top, int left, int bottom, int right){
+		ObjectInstance o = s.getObjectsOfTrueClass(CLASSDOOR).get(i);
+		setRegion(o, top, left, bottom, right);
+	}
+	
+	public static void setRoom(ObjectInstance o, int top, int left, int bottom, int right, String color){
+		setRegion(o, top, left, bottom, right);
+		o.setValue(ATTCOLOR, color);
+	}
+	
+	public static void setRegion(ObjectInstance o, int top, int left, int bottom, int right){
+		o.setValue(ATTTOP, top);
+		o.setValue(ATTLEFT, left);
+		o.setValue(ATTBOTTOM, bottom);
+		o.setValue(ATTRIGHT, right);
+	}
+	
+	protected static void createNInstances(Domain domain, State s, String className, int n){
+		for(int i = 0; i < n; i++){
+			ObjectInstance o = new ObjectInstance(domain.getObjectClass(className), className+i);
+			s.addObject(o);
+		}
+	}
+	
+	
+	public static int maxRoomXExtent(State s){
+		
+		int max = 0;
+		List <ObjectInstance> rooms = s.getObjectsOfTrueClass(CLASSROOM);
+		for(ObjectInstance r : rooms){
+			int right = r.getDiscValForAttribute(ATTRIGHT);
+			if(right > max){
+				max = right;
+			}
+		}
+		
+		return max;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 319:328, 333:342.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.